src/hdmf/common/table.py

```diff
@@ -708,19 +681,7 @@ def add_column(self, **kwargs):  # noqa: C901
         :raises ValueError: if the column has already been added to the table
         """
         name, data = getargs('name', 'data', kwargs)
-        index, table, enum, col_cls, term_set= popargs('index', 'table', 'enum', 'col_cls', 'term_set', kwargs)
-
-        if term_set is not None:
-            bad_data = []
-            for val in data:
-                if term_set.validate(term=val):
-                    continue
-                else:
-                    bad_data.append(val)
-            if len(bad_data)!=0:
-                bad_data_string = str(bad_data)[1:-1]
-                msg = ("%s is not in the term set." % bad_data_string)
-                raise ValueError(msg)
```

Comment:
Same here. Either all the values get added or an error is raised and none of them get added. This means that data will be validated twice, but I think that's OK.

Reply:
Agreed. Will do

Reply:
Alternative, we could "roll back" the new additions, but I think that would be messy.